html5/kubeinvaders.js

```diff
@@ -199,77 +217,75 @@ function currentChaosContainerJsonTextAreaVal() {
     return editor_chaos_container_definition.getValue();
 }
 
-function getCodeName() {
-    const prefixes = ['astro', 'cosmo', 'space', 'star', 'nova', 'nebula', 'galaxy', 'super', 'hyper', 'quantum'];
-    const suffixes = ['nova', 'tron', 'wave', 'core', 'pulse', 'jump', 'drive', 'ship', 'gate', 'hole'];
+function showErrorAlert(message, newCodename) {
+    $('#alert_placeholder').replaceWith(alert_div + message + '</div>');
+    codename = newCodename;
+}
 
-    const prefix = prefixes[Math.floor(Math.random() * prefixes.length)];
-    const suffix = suffixes[Math.floor(Math.random() * suffixes.length)];
+async function setCodeNameToTextInput(elementId) {
+    const response = await httpGetToK8S("/codename");
 
-    const   
-    codename = prefix + suffix;
-    return codename;
-}
+    if (!response.ok) {
+        showErrorAlert('Error getting codename from backend. ', "error_fix_getcodename_from_backend");
+        return;
+    }
 
-function setCodeNameToTextInput(elementId) {
-    var oReq = new XMLHttpRequest();
-    oReq.onreadystatechange = function () {
-        if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
-            codename = this.responseText.trim();
-            $("#" + elementId).val(codename);
-            $("#" + elementId).text(codename);
-            if (codename == "") {
-                $('#alert_placeholder').replaceWith(alert_div + 'Error getting codename from backend. </div>');
-                codename = "error_fix_getcodename_from_backend";
-            }
-        }
-    };;
-    oReq.open("GET", k8s_url + "/codename");
-    oReq.send();
+    const value = (await response.text()).trim();
+
+    $("#" + elementId).val(value);
+    $("#" + elementId).text(value);
+
+    if (value == "") {
+        showErrorAlert('Error getting codename from backend. ', "error_fix_getcodename_from_backend");
+    } else {
+        codename = value;
+    }
 }
 
-function getMetrics() {
-    var oReq = new XMLHttpRequest();
-    oReq.onload = function () {
-        var lines = this.responseText.split('\n');
-        for (var i = 0;i < lines.length;i++){
-            metric = lines[i].split(' ');
+async function getMetrics() {
+    const response = await httpGetToK8S("/metrics");
```

Comment:
Switched from `XMLHttpRequest` to `httpGetToK8S`

Reply:
Also used [async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to not rely on callbacks and make the code seamless syncronous.